src/main/java/com/google/sps/servlets/StartGameInstanceServlet.java

```diff
@@ -29,15 +47,43 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
     GameInstanceDao dao = (GameInstanceDao) this.getServletContext().getAttribute("gameInstanceDao");
     GameInstance newRoom = dao.getGameInstance(roomId);
-
+        
     if(newRoom == null){
         response.setStatus(404);
         response.getWriter().println("Error, not found.");
         return;
     }
-    
-    // Activate room
-    newRoom.setIsActive(true);
+    if (action.equals("end")) {
+        newRoom.setIsActive(false);
+        newRoom.setCurrentQuestion(null);
+    }
+    if (action.equals("start")) {
+        ApiFuture<QuerySnapshot> future = db.collection("games").document(newRoom.getGameId()).collection("questions").get();
```

Comment:
Are questions always returned in the proper sorted order?

Reply:
> TODO in a followup PR: Separate each action into its own servlet and use POST requests for state changes.

Yeah, did this a little hurried because of the demo presentation but will do!

Reply:
I don't know but I think I can sort them by creation date? But I think rodrigo will post the questions all at once, not sure how to do it, do you have an idea?

Reply:
So according to the docs a query returns documents in ascending order by document ID. So if the doc IDs are set up that way it should work as you have it now. But if not, there could be a field for index - then in the query add `.orderBy("index")`. Or (maybe a little less friendly approach) store `nextQuestionId` and maybe `previousQuestionId` in a question doc to form a linked list.